extensions/amp-ad-network-smartadserver-impl/0.1/amp-ad-network-smartadserver-impl.js

```diff
@@ -180,6 +181,22 @@ export class AmpAdNetworkSmartadserverImpl extends AmpA4A {
     };
   }
 
+  /**
+   * Adds message event listener and triggers collapsing
+   */
+  addListener() {
+    const messageListener = (event) => {
+      if (
+        event.data.sentinel === this.sentinel &&
+        event.data.type === 'collapse'
+      ) {
+        this./*OK*/ collapse();
```

Comment:
Sorry, I meant to say `this.attemptCollapse();` cause it'll determine when you are allowed to collapse without shifting the layout.

Reply:
Sorry I think you might have misunderstood. We explicitly do not allow collapsing while in the viewport. You are free to collapse if the ad is below the viewport. If I remember correctly we discussed some strategies to support this server side?

Reply:
Hello guys 👋 
Sorry to interrupt. @calebcordry what is the common use case when no ad is delivered on the placement but the later is above the viewport? Do usually ad networks keep the blank space in that case?
Thanks!

Reply:
Yes that is right the blank space is kept so that the content isn't shifted in front of the user. Here is an example of an ad that was resized but you can see the space is preserved.
![image](https://user-images.githubusercontent.com/16087874/161797663-01950614-19dc-4cfa-ac6b-c22c0f8feb54.png)